app/services/BusinessService.scala

```diff
@@ -62,7 +62,24 @@ class BusinessServiceImpl @Inject() (businessConnector: IFSBusinessDetailsConnec
       dateOfBirth    <- EitherT.fromEither[Future](citizenDetails.parseDoBToLocalDate)
     } yield dateOfBirth
 
-  def getBusinessIncomeSourcesSummary(taxYear: TaxYear, nino: Nino, businessId: BusinessId)(implicit
-      hc: HeaderCarrier): ApiResultT[BusinessIncomeSourcesSummaryResponse] =
-    businessConnector.getBusinessIncomeSourcesSummary(taxYear, nino, businessId)
+  def getAllBusinessIncomeSourcesSummaries(taxYear: TaxYear, nino: Nino)(implicit
+      hc: HeaderCarrier): ApiResultT[List[BusinessIncomeSourcesSummaryResponse]] = {
+
+    def getSummaryForEachBusinessId(remainingIds: List[String],
+                                    summaries: List[BusinessIncomeSourcesSummaryResponse]): ApiResultT[List[BusinessIncomeSourcesSummaryResponse]] =
+      remainingIds match {
+        case Nil => EitherT.rightT(summaries)
+        case head :: tail =>
+          for {
+            newSummary       <- businessConnector.getBusinessIncomeSourcesSummary(taxYear, nino, head)
+            updatedSummaries <- getSummaryForEachBusinessId(tail, summaries :+ newSummary)
```

Comment:
Using recurrency is not common in Scala (it is common in scala courses, but not in business code), few reasons:
- It is hard to make it right, small mistakes and you may have an inifite loop
- it is hard to read/understand
- this one is not even a `@tailrec` which is not optimized by compiler, so it will be very slow and heave memory use

Luckily, we don't need to use it here. Cats bring a little helper function `traverse`. Try:

```
  for {
    businesses <- getBusinesses(nino)
    businessIds = businesses.map(_.businessId)
    summaryList <- businessIds.traverse { businessId =>
      businessConnector.getBusinessIncomeSourcesSummary(taxYear, nino, businessId)
    }
  } yield summaryList
}
```